samples/client/src/main/java/global/covesa/sdk/client/MainActivity.kt

```diff
@@ -27,5 +33,14 @@ class MainActivity : ComponentActivity() {
                 MainUi(viewModel)
             }
         }
+        subscribeActions()
+    }
+
+    private fun subscribeActions() {
+        CoroutineScope(Dispatchers.IO).launch {
+            ActionEvent.events.collect {
+                it.handleAction(this@MainActivity)
```

Comment:
You are giving a reference to an object (`MainActivity`) which may be destroyed to a global event queue, which may try to use it when it's already destroyed. We need some common scope for these.